### PR TITLE
148656-agent-system-test via agent-installer

### PR DESCRIPTION
**Pull request by nagaraj-kandoor**

### CODE REVIEW

**Review by FelixMarxIBM — COMMENTED** [review]
@nagaraj-kandoor Looks good,
now also do this for 
https://github.com/instana/instana-agent-docker/pull/192/files#diff-3d0daaf36f9d9e30f5d9d1a4b42bb0665a4980c4f292a6de164635393c7da29dR728
and
https://github.com/instana/instana-agent-docker/pull/192/files#diff-3d0daaf36f9d9e30f5d9d1a4b42bb0665a4980c4f292a6de164635393c7da29dR655

After that, we can delete the
`create-verification-vm-task.yml`, `destroy-verification-vm-task.yml` and `verify-e2e-task.yml` files

File: pipeline.yml [hunk 1]
```diff
@@ -618,49 +626,31 @@ jobs:
...
+        - get: instana-agent-docker-git
+          passed: [ amd64-build ]
+        - get: agent-version
+          passed: [ amd64-build ]
+        - get: agent-static-amd64
+          params:
+            format: oci
+          passed: [amd64-build]
+          trigger: true
+      - put: metadata
```
Comment (FelixMarxIBM) [inline pipeline.yml:640]: I think the `- put: metadata` step can also be done in the `- in_parallel:` block

Comment (nagaraj-kandoor) [reply pipeline.yml]: fixed

File: pipeline.yml [hunk 2]
```diff
@@ -618,187 +626,108 @@ jobs:
...
         vars:
-          instana-agent-image: gcr.io/instana-agent-qa/instana-agent-docker@((.:digest))
-          instana-agent-jdk: OpenJ9
-          instana-api-key: ((qa-instana-api-token))
-          instana-zone: agent-verification
-          instana-host: agent-verification
-          instance-name: instana-agent-amd64-static-j9
-          gcp-project: instana-agent-qa
-          gcp-zone: europe-west3-c
+          gcp-labels: 'jvm=azul,container_type=static'
```
Comment (FelixMarxIBM) [inline pipeline.yml:674]:
```suggestion
          gcp-labels: 'jvm=j9,container_type=static'
```

File: pipeline.yml [hunk 3]
```diff
@@ -618,187 +626,108 @@ jobs:
...
-        vars:
-          instana-agent-image: gcr.io/instana-agent-qa/instana-agent-docker@((.:digest))
-          instana-agent-jdk: OpenJ9
-          instana-api-key: ((qa-instana-api-token))
-          instana-zone: agent-verification
-          instana-host: agent-verification
-          instance-name: instana-agent-amd64-dynamic-j9
-          gcp-project: instana-agent-qa
-          gcp-zone: europe-west3-c
+          gcp-labels: 'jvm=azul,container_type=dynamic'
```
Comment (FelixMarxIBM) [inline pipeline.yml:726]:
```suggestion
          gcp-labels: 'jvm=j9,container_type=dynamic'
```

File: pipeline.yml (hunk 2 above)
Comment (nagaraj-kandoor) [reply pipeline.yml]: fixed

File: pipeline.yml (hunk 3 above)
Comment (nagaraj-kandoor) [reply pipeline.yml]: fixed

File: pipeline.yml [hunk 4]
```diff
@@ -616,203 +624,229 @@ jobs:
...
+        passed: [ ppc64le-build, s390x-build, arm64-build, amd64-build, twistlock-security-scan, verify-amd64-static-docker-j9-attachment, verify-amd64-static-docker-hotspot-attachment, verify-amd64-static-j9-docker-j9-attachment, verify-amd64-static-j9-docker-hotspot-attachment, verify-amd64-dynamic-docker-j9-attachment, verify-amd64-dynamic-docker-hotspot-attachment, verify-amd64-dynamic-j9-docker-j9-attachment, verify-amd64-dynamic-j9-docker-hotspot-attachment ]
         trigger: true
       - get: agent-dynamic-amd64
-        passed: [ verify-amd64-dynamic ]
+        passed: [ verify-amd64-dynamic-docker-hotspot-attachment ]
         params:
           skip_download: true
       - get: agent-dynamic-amd64-j9
-        passed: [ verify-amd64-dynamic-j9 ]
+        passed: [ verify-amd64-dynamic-j9-docker-j9-attachment]
```
Comment (FelixMarxIBM) [inline pipeline.yml:849]:
```suggestion
        passed: [ verify-amd64-dynamic-j9-docker-j9-attachment, verify-amd64-dynamic-j9-docker-hotspot-attachment ]
```

File: pipeline.yml [hunk 5]
```diff
@@ -833,11 +867,11 @@ jobs:
           skip_download: true
 
       - get: agent-static-amd64
-        passed: [ verify-amd64-static ]
+        passed: [ verify-amd64-static-docker-hotspot-attachment ]
```
Comment (FelixMarxIBM) [inline pipeline.yml:870]:
```suggestion
        passed: [ verify-amd64-static-docker-j9-attachment, verify-amd64-static-docker-hotspot-attachment ]
```

File: pipeline.yml [hunk 6]
```diff
@@ -833,11 +867,11 @@ jobs:
...
         params:
           skip_download: true
       - get: agent-static-amd64-j9
-        passed: [ verify-amd64-static-j9 ]
+        passed: [ verify-amd64-static-j9-docker-j9-attachment ]
```
Comment (FelixMarxIBM) [inline pipeline.yml:874]:
```suggestion
        passed: [ verify-amd64-static-j9-docker-j9-attachment, verify-amd64-static-j9-docker-hotspot-attachment ]
```

File: pipeline.yml [hunk 7]
```diff
@@ -616,203 +624,229 @@ jobs:
...
       - get: agent-version
-        passed: [ ppc64le-build, s390x-build, arm64-build, amd64-build, twistlock-security-scan, verify-amd64-static, verify-amd64-dynamic, verify-amd64-static-j9, verify-amd64-dynamic-j9 ]
+        passed: [ ppc64le-build, s390x-build, arm64-build, amd64-build, twistlock-security-scan, verify-amd64-static-docker-j9-attachment, verify-amd64-static-docker-hotspot-attachment, verify-amd64-static-j9-docker-j9-attachment, verify-amd64-static-j9-docker-hotspot-attachment, verify-amd64-dynamic-docker-j9-attachment, verify-amd64-dynamic-docker-hotspot-attachment, verify-amd64-dynamic-j9-docker-j9-attachment, verify-amd64-dynamic-j9-docker-hotspot-attachment ]
         trigger: true
       - get: agent-dynamic-amd64
-        passed: [ verify-amd64-dynamic ]
+        passed: [ verify-amd64-dynamic-docker-hotspot-attachment ]
```
Comment (FelixMarxIBM) [inline pipeline.yml:845]:
```suggestion
        passed: [ verify-amd64-dynamic-docker-j9-attachment, verify-amd64-dynamic-docker-hotspot-attachment ]
```

File: pipeline.yml (hunk 5 above)
Comment (nagaraj-kandoor) [reply pipeline.yml:878]: fixed.

**Review by FelixMarxIBM — APPROVED** [review]
LGTM, nice work

**Pull request merged**